### PR TITLE
Cleanup Slot Verification

### PR DESCRIPTION
**Pull request by vazois**
- [x] Refactor object commands to use cluster slot verification
- [x] Add comments to slot verification testing
- [x] Fix calculation of keynum for commands containing number of keys as part of the parameters

### CODE REVIEW

File: libs/server/Resp/RespServerSessionSlotVerify.cs [hunk 1]
```diff
@@ -65,9 +55,10 @@ bool CanServeSlot(RespCommand cmd)
                             csvi.step = findRange.KeyStep;
                             break;
                         case FindKeysKeyNum:
+                            var beginSearch = (BeginSearchIndex)specs[0].BeginSearch;
                             var findKeysKeyNum = (FindKeysKeyNum)specs[0].FindKeys;
                             csvi.firstKey = searchIndex.Index + findKeysKeyNum.FirstKey - 1;
-                            csvi.lastKey = searchIndex.Index + parseState.GetInt(0);
+                            csvi.lastKey = searchIndex.Index + parseState.GetInt(beginSearch.Index + findKeysKeyNum.KeyNumIdx - 1);
```
Comment (badrishc) [inline libs/server/Resp/RespServerSessionSlotVerify.cs:61]: this is wrong, need to merge from the recently merged PR.

File: libs/server/Resp/RespServerSessionSlotVerify.cs [hunk 2]
```diff
@@ -65,9 +55,10 @@ bool CanServeSlot(RespCommand cmd)
                             csvi.step = findRange.KeyStep;
                             break;
                         case FindKeysKeyNum:
+                            var beginSearch = (BeginSearchIndex)specs[0].BeginSearch;
```
Comment (badrishc) [inline libs/server/Resp/RespServerSessionSlotVerify.cs:58]: Why is there a beginSearch and a searchIndex above?

File: libs/server/Resp/RespServerSessionSlotVerify.cs (hunk 1 above)
Comment (badrishc) [reply libs/server/Resp/RespServerSessionSlotVerify.cs:61]: This line should be: `csvi.lastKey = csvi.firstKey + parseState.GetInt(searchIndex.Index + findKeysKeyNum.KeyNumIdx - 1);`

Comment (vazois) [reply libs/server/Resp/RespServerSessionSlotVerify.cs]: Fixed the issue of wrong merge

File: libs/server/Resp/RespServerSessionSlotVerify.cs (hunk 2 above)
Comment (vazois) [reply libs/server/Resp/RespServerSessionSlotVerify.cs]: Removed this because it is not needed. Forgot that searchIndex was available.

**Review by badrishc — APPROVED** [review]

**Pull request merged**